### ISSUE TITLE
Does not work if there is no `docker` group

### ISSUE THREAD

**Issue by MikeSpreitzer** [Mirantis/cri-dockerd#48]
I just installed Ubuntu 22.04 server on a new VM.  The install process offered to "install" "docker", and I took the offer.  Later I found out this install was done with `snap`.  It turns out that `snap install docker`, at least on Ubuntu 22.04, does _NOT_ create a usergroup named `docker`.  But the `cri-docker.socket` systemd unit here assumes the existence of such a group.

**Comment by afbjorklund** [comment]
I don't think it has been tested with anything, except for the regular installation from a deb.

https://docs.docker.com/engine/install/ubuntu/

It might be supported with "docker.io", not sure ?

https://packages.ubuntu.com/jammy/docker.io

**Comment by afbjorklund** [comment]
If it comes to it, it would be possible to remove "cri-docker.socket" and modify "cri-docker.service" to run without it.

That would make `cri-dockerd` start on boot, and not on demand, but might work better with odd docker installations.

`Requires=cri-docker.socket`

`ExecStart=/usr/bin/cri-dockerd --container-runtime-endpoint fd:// --network-plugin=`

**Comment by folarinmartins** [comment]
@evol262 Good point. That's an issue you can raise over at snap

**Comment by criscola** [comment]
FYI this happened to me as well on ubuntu 20.04, I had to add group docker

**Comment by afbjorklund** [comment]
How did you install docker? The packages _should_ have added the "docker" group

**Comment by criscola** [comment]
Should have mentioned I didn't yet install it :) probably makes sense to do that before installing the cri.

**Comment by afbjorklund** [comment]
It seems that the cri-dockerd packages have a dependency on **containerd** instead of dockerd...

**Comment by criscola** [comment]
That clarifies thanks @evol262
